[CI] KeyRotationIT testEachInstallOrRotationAddsExactlyOneNewKey failing

**Build Scans:**
- [elasticsearch-periodic #12592 / openjdk26_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/rrh4ppqousscc)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:encryption:internalClusterTest" --tests "org.elasticsearch.xpack.encryption.KeyRotationIT.testEachInstallOrRotationAddsExactlyOneNewKey" -Dtests.seed=6B10635557D93BCF -Dtests.locale=gsw-LI -Dtests.timezone=America/Port-au-Prince -Druntime.java=26
```

**Applicable branches:**
9.5

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.encryption.KeyRotationIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testEachInstallOrRotationAddsExactlyOneNewKey'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was not completed within the timeout
```

**Issue Reasons:**
- [9.5] 1 consecutive failures in test testEachInstallOrRotationAddsExactlyOneNewKey
- [9.5] 1 consecutive failures in step openjdk26_checkpart2_java-matrix
- [9.5] 1 consecutive failures in pipeline elasticsearch-periodic

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-core-infra.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -766,9 +766,6 @@ tests:
 - class: org.elasticsearch.xpack.transform.integration.continuous.TransformContinuousIT
   method: testContinuousEvents
   issue: https://github.com/elastic/elasticsearch/issues/153782
-- class: org.elasticsearch.xpack.encryption.KeyRotationIT
-  method: testEachInstallOrRotationAddsExactlyOneNewKey
-  issue: https://github.com/elastic/elasticsearch/issues/153024
 - class: org.elasticsearch.xpack.stateless.cache.SharedBlobCacheWarmingServiceTests
   method: testPopulateCacheWithSharedSourceInputStreamFactory
   issue: https://github.com/elastic/elasticsearch/issues/154597

--- a/x-pack/plugin/encryption/src/internalClusterTest/java/org/elasticsearch/xpack/encryption/KeyRotationIT.java
+++ b/x-pack/plugin/encryption/src/internalClusterTest/java/org/elasticsearch/xpack/encryption/KeyRotationIT.java
@@ -7,8 +7,6 @@
 package org.elasticsearch.xpack.encryption;
 
 import org.elasticsearch.TransportVersion;
-import org.elasticsearch.action.admin.cluster.tasks.PendingClusterTasksRequest;
-import org.elasticsearch.action.admin.cluster.tasks.TransportPendingClusterTasksAction;
 import org.elasticsearch.cluster.AbstractNamedDiffable;
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.cluster.ClusterStateUpdateTask;
@@ -51,7 +49,6 @@ import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Function;
 
 import static org.hamcrest.Matchers.anEmptyMap;
-import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
 import static org.hamcrest.Matchers.hasItem;
@@ -80,22 +77,7 @@ public class KeyRotationIT extends SecurityIntegTestCase {
         for (String nodeName : internalCluster().getNodeNames()) {
             internalCluster().getInstance(KeyRotationCoordinator.class, nodeName).close();
         }
-        assertBusy(
-            () -> assertThat(
-                client().execute(TransportPendingClusterTasksAction.TYPE, new PendingClusterTasksRequest(TEST_REQUEST_TIMEOUT))
-                    .get()
-                    .pendingTasks()
-                    .stream()
-                    .filter(t -> {
-                        String src = t.getSource().string();
-                        return src.contains("project-encryption-key") || src.startsWith("re-encrypt-");
-                    })
-                    .toList(),
-                empty()
-            ),
-            30,
-            TimeUnit.SECONDS
-        );
+        waitNoPendingTasksOnAll();
     }
 
     @Override

